### PR TITLE
feat: Add batch convert project, optimize project list query.

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Add batch convert project, optimize project list query.

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMjg2LDEyOTcsMTI4MCwxMjUwLDEyMjQsMTIxOCwxMTkwXSwiYXNzaWduZWUiOlsiMTAwMTA3MyJdfQ%3D%3D&id=317965&iterationID=1280&pId=0&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   Add batch convert project, optimize project list query.           |
| 🇨🇳 中文    |      优化项目列表查询        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/core/legacy/dao/application.go [hunk 1]
```diff
@@ -98,6 +98,20 @@ func (client *DBClient) GetApplicationCountByProjectID(projectID int64) (int64,
 	return total, nil
 }
 
+type AppCount struct {
+	ID    int64
```
Comment (sfwn) [inline internal/core/legacy/dao/application.go:102]: confusing var definition `ID` here.
use `ProjectID` at least.

File: internal/core/legacy/dao/application.go [hunk 2]
```diff
@@ -98,6 +98,20 @@ func (client *DBClient) GetApplicationCountByProjectID(projectID int64) (int64,
...
+	Count int
+}
+
+func (client *DBClient) GetApplicationCountByProjectIDs(projectIDs []int64) ([]AppCount, error) {
+	var counts []AppCount
+	if err := client.Debug().Table("dice_app").Select("project_id as id, COUNT(id) as count").Where("project_id IN (?)", projectIDs).
```
Comment (sfwn) [inline internal/core/legacy/dao/application.go:108]: any `TableName` func for `dice_app`

File: internal/core/legacy/dao/member.go [hunk 3]
```diff
@@ -343,6 +343,20 @@ func (client *DBClient) GetMembersWithoutExtraByScope(scopeType apistructs.Scope
 	return total, members, nil
 }
 
+type MemberCount struct {
+	ID    int64
```
Comment (sfwn) [inline internal/core/legacy/dao/member.go:347]: same

File: internal/core/legacy/services/project/project.go [hunk 4]
```diff
@@ -1282,6 +1277,78 @@ func (p *Project) convertToProjectDTO(joined bool, project *model.Project) apist
...
+	if err != nil {
+		l.WithError(err).Errorln("failed to count app")
+	}
+	memberCountMap := make(map[int64]int)
+	for _, i := range memberCount {
+		memberCountMap[i.ID] = i.Count
+	}
+	res := make([]apistructs.ProjectDTO, 0, len(projects))
+	for _, project := range projects {
+		var rollbackConfig map[string]int
```
Comment (sfwn) [inline internal/core/legacy/services/project/project.go:1306]: single `convertToProjectDTO` is needed, just pass in `ProjectStats`.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/core/legacy/dao/application.go (hunk 1 above)
Comment (shuofan) [reply internal/core/legacy/dao/application.go]: done

File: internal/core/legacy/dao/application.go (hunk 2 above)
Comment (shuofan) [reply internal/core/legacy/dao/application.go]: done

File: internal/core/legacy/dao/member.go (hunk 3 above)
Comment (shuofan) [reply internal/core/legacy/dao/member.go]: done

File: internal/core/legacy/services/project/project.go (hunk 4 above)
Comment (shuofan) [reply internal/core/legacy/services/project/project.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**